pkg/scheduler/framework/plugins/requestedtocapacityratio/requested_to_capacity_ratio.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requestedtocapacityratio
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+// Name of this plugin.
+const Name = "RequestedToCapacityRatio"
+
+// Args holds the args that are used to configure the plugin.
+type Args struct {
+	FunctionShape       priorities.FunctionShape
+	ResourceToWeightMap priorities.ResourceToWeightMap
+}
+
+// New initializes a new plugin and returns it.
+func New(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
+	args := &Args{}
+	if err := framework.DecodeInto(plArgs, args); err != nil {
+		return nil, err
+	}
+	p := priorities.RequestedToCapacityRatioResourceAllocationPriority(args.FunctionShape, args.ResourceToWeightMap)
+	return &RequestedToCapacityRatio{
+		handle:     handle,
+		prioritize: p.PriorityMap,
+	}, nil
+}
+
+// RequestedToCapacityRatio is a score plugin that allow users to apply bin packing
+// on core resources like CPU, Memory as well as extended resources like accelerators.
+type RequestedToCapacityRatio struct {
+	handle     framework.FrameworkHandle
+	prioritize priorities.PriorityMapFunction
+}
+
+var _ framework.ScorePlugin = &RequestedToCapacityRatio{}
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *RequestedToCapacityRatio) Name() string {
+	return Name
+}
+
+// Score invoked at the score extension point.
+func (pl *RequestedToCapacityRatio) Score(ctx context.Context, _ *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	nodeInfo, exist := pl.handle.NodeInfoSnapshot().NodeInfoMap[nodeName]
```

Comment:
use SnapshotSharedLister to get the nodeInfo, rebase and see how other plugins get the nodeInfo.